patches/api/0416-Brigadier-based-command-API.patch

```diff
@@ -0,0 +1,758 @@
+      * @throws IllegalStateException if no biome by the given key is registered.
+      */
+     void setBiomeKey(RegionAccessor accessor, int x, int y, int z, NamespacedKey biomeKey);
++
++    java.util.Map<String, org.bukkit.command.Command> getCommandMirrorMap();
+     // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/command/Command.java b/src/main/java/org/bukkit/command/Command.java
+index b791358f90fe92bc2264d9a26492245763813af3..28fc568a033f0c8c58292983b5b5f2adec1987c9 100644
+--- a/src/main/java/org/bukkit/command/Command.java
++++ b/src/main/java/org/bukkit/command/Command.java
+@@ -488,4 +488,12 @@ public abstract class Command {
+     public String toString() {
+         return getClass().getName() + '(' + name + ')';
+     }
++
++    // Paper start
++    @org.jetbrains.annotations.ApiStatus.Internal
++    public boolean canBeOverriden() { return false; }
++
++    @org.jetbrains.annotations.ApiStatus.Internal
++    public String registeredNamespace;
++    // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/command/CommandMap.java b/src/main/java/org/bukkit/command/CommandMap.java
+index 864c263bbd4dd6dd7c37a74b39b1a40a884d0731..88d92350b59333c2148190c5e4974f254b0139a9 100644
+--- a/src/main/java/org/bukkit/command/CommandMap.java
++++ b/src/main/java/org/bukkit/command/CommandMap.java
+@@ -5,6 +5,7 @@ import org.bukkit.Location;
+ import org.jetbrains.annotations.NotNull;
+ import org.jetbrains.annotations.Nullable;
+ 
++@Deprecated(forRemoval = true) // Paper
+ public interface CommandMap {
+ 
+     /**
+diff --git a/src/main/java/org/bukkit/command/FormattedCommandAlias.java b/src/main/java/org/bukkit/command/FormattedCommandAlias.java
+index 9d4f553c04784cca63901a56a7aea62a5cae1d72..abe256e1e45ce28036da4aa1586715bc8a1a3414 100644
+--- a/src/main/java/org/bukkit/command/FormattedCommandAlias.java
++++ b/src/main/java/org/bukkit/command/FormattedCommandAlias.java
+@@ -117,7 +117,7 @@ public class FormattedCommandAlias extends Command {
+             index = formatString.indexOf('$', index);
+         }
+ 
+-        return formatString;
++        return formatString.trim(); // Paper - Causes an extra space at the end, breaks with brig commands
+     }
+ 
+     @NotNull
+diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
+index ac9a28922f8a556944a4c3649d74c32c622f0cb0..6abf192bf7d3102a72595e5b8090a13b88ab30f7 100644
+--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
++++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
+@@ -21,11 +21,20 @@ import org.bukkit.util.StringUtil;
+ import org.jetbrains.annotations.NotNull;
+ import org.jetbrains.annotations.Nullable;
+ 
++@Deprecated // Paper
+ public class SimpleCommandMap implements CommandMap {
+-    protected final Map<String, Command> knownCommands = new HashMap<String, Command>();
++    protected Map<String, Command> knownCommands; // Paper
```

Comment:
I don't see the use of the removed final? It looks like it's just assigned in the constructor as normal.